src/makeCalculator.js

```diff
@@ -37,7 +37,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    add(a, b) {
+      this.result = a + b;
+
+      return this.result;
+    },
+    subtract(a, b) {
+      this.result = a - b;
+
+      return this.result;
+    },
+    multiply(a, b) {
+      this.result = a * b;
+
+      return this.result;
+    },
+    divide(a, b) {
+      this.result = a / b;
+
+      return this.result;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, num) {
+      this.result = callback.call(this, this.result, num);
+
+      return this;
+    },
```

Comment:
no need to path this.result to function as argument, you have the access to the context inside the function when you call it with `.call` and your context